x-pack/plugins/observability/public/pages/rules/components/last_response_filter.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable react/function-component-definition */
+
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiFilterGroup,
+  EuiPopover,
+  EuiFilterButton,
+  EuiFilterSelectItem,
+  EuiHealth,
+} from '@elastic/eui';
+import { AlertExecutionStatuses, AlertExecutionStatusValues } from '../../../../../alerting/common';
+import { getHealthColor, rulesStatusesTranslationsMapping } from '../config';
+import { StatusFilterProps } from '../types';
+
+export const LastResponseFilter: React.FunctionComponent<StatusFilterProps> = ({
+  selectedStatuses,
+  onChange,
+}: StatusFilterProps) => {
+  const [selectedValues, setSelectedValues] = useState<string[]>(selectedStatuses);
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(selectedValues);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedValues]);
+
+  useEffect(() => {
```

Comment:
do you really need `selectedValues` as a state? it seems like you can do without it